site/src/pages/TemplateVersionPage/TemplateVersionPage.tsx

```diff
@@ -0,0 +1,198 @@
+import { makeStyles } from "@material-ui/core/styles"
+import { useMachine } from "@xstate/react"
+import { MarkdownIcon } from "components/Icons/MarkdownIcon"
+import { TerraformIcon } from "components/Icons/TerraformIcon"
+import { Loader } from "components/Loader/Loader"
+import { Margins } from "components/Margins/Margins"
+import {
+  PageHeader,
+  PageHeaderCaption,
+  PageHeaderTitle,
+} from "components/PageHeader/PageHeader"
+import { Stack } from "components/Stack/Stack"
+import { Stats, StatsItem } from "components/Stats/Stats"
+import { SyntaxHighlighter } from "components/SyntaxHighlighter/SyntaxHighlighter"
+import { useOrganizationId } from "hooks/useOrganizationId"
+import { Helmet } from "react-helmet-async"
+import { Link, useParams, useSearchParams } from "react-router-dom"
+import { combineClasses } from "util/combineClasses"
+import { createDayString } from "util/createDayString"
+import { pageTitle } from "util/page"
+import { templateVersionMachine } from "xServices/templateVersion/templateVersionXService"
+
+type Params = {
+  version: string
+  template: string
+}
+
+/**
+ * Param used in the query string to locate the open file
+ */
+const FileTabParam = "file"
+
+const useFileTab = () => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const value = searchParams.get(FileTabParam)
+
+  return {
+    value: value ? Number(value) : 0,
+    set: (value: number) => {
+      searchParams.set(FileTabParam, value.toString())
+      setSearchParams(searchParams)
+    },
+  }
+}
+
+export const TemplateVersionPage: React.FC = () => {
+  const { version: versionName, template } = useParams() as Params
+  const orgId = useOrganizationId()
+  const [state] = useMachine(templateVersionMachine, {
+    context: { versionName, orgId },
+  })
+  const { files, version } = state.context
+  const fileTab = useFileTab()
+  const styles = useStyles()
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle(`Version ${versionName} · ${template}`)}</title>
+      </Helmet>
+
+      <Margins>
+        <PageHeader>
+          <PageHeaderCaption>Versions</PageHeaderCaption>
+          <PageHeaderTitle>{versionName}</PageHeaderTitle>
+        </PageHeader>
+      </Margins>
+
+      {!files && !state.matches("done.error") && <Loader />}
+
+      {state.matches("done.ok") && version && files && (
```

Comment:
looks like there are some cases where neither the loader nor the data will render. Should it render an error or anything in that case?

Reply:
Yes!